code/mob/living/critter/ai.dm

```diff
@@ -217,20 +228,50 @@ var/list/ai_move_scheduled = list()
 		holder = null
 		..()
 
-	///Called when the task is switched to by the holder
+	/// Called when the task is switched to by the holder
 	proc/switched_to()
 
+	/// Called on every mobAI tick - this tick rate is determined by mobAI priority
 	proc/on_tick()
 
+	/// Called on every mobAI tick - returns a task to switch to, ending this task, or null to continue
 	proc/next_task()
 		return null
 
+	/// Called whenever the task is started or ended. Override this instead of reset()
 	proc/on_reset()
 		holder.target = null
 
-	proc/evaluate() // evaluate the current environment and assign priority to switching to this task
+	/// Evaluate the current environment and assign priority to switching to this task
+	proc/evaluate()
 		return 0
 
+	/// Returns a list of atoms that are potential targets for this task
+	proc/get_targets()
+		return list()
+
+	/// Takes a list of atoms which are then evaluated, before setting the holder's target. Note this checks a path exists to each target
+	proc/get_best_target(list/atom/targets)
+		. = null
+		var/best_score = -1.#INF
+		var/list/best_path = null
+		if(length(targets))
+			for(var/atom/A in targets)
```

Comment:
This is actually cheaper in 99% of cases, since the targets list is usually populated by `view()` and `score_target()` is based on distance from the holder, they're already sorted on score by default. From profiling nearly every single call to `get_best_target()` results in only one call to `get_path_to()`